File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hash;
        for(int i = 0; i < nums.size(); i++) {
            int v = target - nums[i];
            auto it = hash.find(v);
            if (it != hash.end() && it->second != i) {
                return {it->second, i};
            }
            hash[nums[i]] = i;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    hash := make(map[int]int)
    for i, v := range nums {
        x := target - v
        index, ok := hash[x]
        if ok {
            return []int{index, i}
        }
        hash[v] = i
    }
    return []int{}
}